src/Staking.sol

```diff
@@ -277,12 +294,21 @@ contract StakeRegistry is AccessControl, Pausable {
         return stakes[_owner].overlay;
     }
 
+    /**
+     * @dev Returns the currently height of the address.
+     * @param _owner address of node
+     */
+    function heightOfAddress(address _owner) public view returns (uint8) {
+        return stakes[_owner].height;
+    }
+
     function calculateEffectiveStake(
         uint256 committedStake,
-        uint256 potentialStakeBalance
+        uint256 potentialStakeBalance,
+        uint8 height
     ) internal view returns (uint256) {
         // Calculate the product of committedStake and unitPrice to get price in BZZ
-        uint256 committedStakeBzz = committedStake * OracleContract.currentPrice();
+        uint256 committedStakeBzz = (2 ** height) * committedStake * OracleContract.currentPrice();
```

Comment:
not,it will be lower in stakeDenisty here because of height

```
uint8 depthResponsibility = _depth - revealedCommit.height;
stakeDensity: revealedCommit.stake * uint256(2 ** depthResponsibility),
```

also even if height  is multiplier on first line, finely value used will be lower one as (min) is used so doesn't matter much

```
       uint256 committedStakeBzz = (2 ** height) * committedStake * OracleContract.currentPrice();

        // Return the minimum value between committedStakeBzz and potentialStakeBalance
        if (committedStakeBzz < potentialStakeBalance) {
            return committedStakeBzz;
        } else {
            return potentialStakeBalance;
        }
```